### PR TITLE
chore: replace nyc with c8

### PR DESCRIPTION
**Pull request by XhmikosR**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

https://packagephobia.com/result?p=nyc%2Cc8

1858 -> 1812 packages

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

---

Requires #3933 to be merged first and then I'll rebase

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,7 +14,6 @@ vendor
 # osx
 .DS_STORE
 .vscode
-coverage
```
Comment (XhmikosR) [inline .gitignore:18]: This was duplicate, see line 7

**Comment by XhmikosR** [comment]
@erezrokah this one could go into 8.6.7 too

**Comment by erezrokah** [comment]
> @erezrokah this one could go into 8.6.7 too

This is a non user facing change, so it won't show up in the change log.

We prefer very small releases as it's easier to detect regressions

**Comment by XhmikosR** [comment]
No worries. Do you have any ETA regarding the dep-related PRs?

We should try to update `http-proxy-middleware` and `inquirer` to the latest versions since they look like they are not ESM only and they don't have any further breaking changes apart from dropping Node.js 10.x. Next I'd like to open a PR from my deps branch (minus the package.json bumps https://github.com/netlify/cli/compare/main...XhmikosR:deps

**Comment by erezrokah** [comment]
> No worries. Do you have any ETA regarding the dep-related PRs?

I checked the relevant updates here https://github.com/netlify/cli/issues/986, so we should start seeing those soon

**Comment by XhmikosR** [comment]
Hmm, it seems now renovate tries to update the ESM only packages...

**Comment by erezrokah** [comment]
@XhmikosR, btw we use [Kodiak](https://kodiakhq.com/) to efficiently sync PRs with `main`.

Once a PR is ready to be merged, we add an `automerge` label to it. Then Kodiak queues it to be merged, and sync it with `main` just before merging (to avoid re-running CI multiple times).

**Comment by XhmikosR** [comment]
Cool, I didn't know, I usually just rebase PRs here if there are conflicts, otherwise I leave them as is.

**Review by erezrokah — APPROVED** [review]
Can't really say no to smaller install size 📦

**Pull request merged**